creator-node/src/ipfsAdd.js

```diff
@@ -0,0 +1,255 @@
+    logger.info(`[ipfsClient - ipfsAddNonImages()] onlyHash=${onlyHash} onlyHashDuration=${durationOnlyHashMs}ms ipfsDaemonHash=${ipfsDaemonHash} ipfsDaemonHashDuration=${durationIpfsLatestAddMs}ms isSameHash=${isSameHash}`)
+
+    if (!isSameHash) {
+      throw new Error(`onlyHash=${onlyHash} and ipfsDaemonHash=${ipfsDaemonHash} are not consistent`)
+    }
+  } catch (e) {
+    const errorMsg = `[ipfsClient - ipfsAddNonImages()] Could not add content to ipfs: ${e.toString()}`
+    logger.error(errorMsg)
+    throw new Error(errorMsg)
+  }
+
+  return onlyHash
+}
+
+/**
+ * Wrapper for ipfs.add() with flag to exclusively generate the CID using the only hash logic, or to also add content to the ipfs daemon.
+ * Used for adding images to also generate dir CIDs.
+ * @param {Object} ipfsLatest ipfs instance (should be v43.0.1)
+ * @param {Object[]} content an Object[] with the structure [{ path: string, content: buffer }, ...]
+ * @param {Object?} ipfsConfig ipfs add config options
+ * @param {Object?} logContext
+ * @param {boolean?} enableIPFSAdd flag to add content to ipfs daemon
+ * @returns {Object[]} only hash responses or ipfs daemon responses with the structure [{path: <string>, cid: <string>, size: <number>}]
+*/
+ipfsAdd.ipfsAddImages = async (content, ipfsConfig = {}, logContext = {}, enableIPFSAdd = false) => {
+  const logger = genericLogger.child(logContext)
+
+  const startOnlyHash = hrtime.bigint()
+  const ipfsAddWithoutDaemonResp = await ipfsAdd.ipfsOnlyHashImages(content)
+  const durationOnlyHashMs = (hrtime.bigint() - startOnlyHash) / BigInt(1000000) // convert ns -> ms
+
+  const ipfsAddWithoutDaemonRespStr = JSON.stringify(sortKeys(ipfsAddWithoutDaemonResp))
+
+  if (!enableIPFSAdd) {
+    logger.info(`[ipfsClient - ipfsAddImages()] onlyHash=${ipfsAddWithoutDaemonRespStr} onlyHashDuration=${durationOnlyHashMs}ms`)
+    return ipfsAddWithoutDaemonResp
+  }
+
+  try {
+    if (!ipfsConfig.timeout) {
+      ipfsConfig.timeout = IPFS_ADD_TIMEOUT_MS
+    }
+
+    const startIpfsLatestAdd = hrtime.bigint()
+    let ipfsAddWithDaemonResp = []
+    for await (const resp of ipfsLatest.add(content, ipfsConfig)) {
+      resp.cid = `${resp.cid}`
+      ipfsAddWithDaemonResp.push(resp)
+    }
+    const durationIpfsLatestAddMs = (hrtime.bigint() - startIpfsLatestAdd) / BigInt(1000000) // convert ns -> ms
+
+    const ipfsAddWithDaemonRespStr = JSON.stringify(sortKeys(ipfsAddWithDaemonResp))
+
+    const isSameResp = ipfsAddWithoutDaemonRespStr === ipfsAddWithDaemonRespStr
+    logger.info(`[ipfsClient - ipfsAddImages()] onlyHash=${ipfsAddWithoutDaemonRespStr} onlyHashDuration=${durationOnlyHashMs}ms ipfsAddWithDaemonResp=${ipfsAddWithDaemonRespStr} ipfsAddWithDaemonRespDuration=${durationIpfsLatestAddMs}ms isSameHash=${isSameResp}`)
+
+    if (!isSameResp) {
+      throw new Error(`onlyHash=${ipfsAddWithoutDaemonRespStr} and ipfsAddWithDaemonResp=${ipfsAddWithDaemonRespStr} are not consistent`)
+    }
```

Comment:
same comment here as above in `ipfsAddNonImages`

Reply:
same convo as https://github.com/AudiusProject/audius-protocol/pull/1939/files/3df6b68c2b948c28480396a676716f130f73b919..b6621cae77d46c5447c5dacbe64aac0d46d03226#r739561533, marking as resolved!